data/shaders/opengl/rayleigh.glsl

```diff
@@ -0,0 +1,237 @@
+float height(const in vec3 orig, const in vec3 center)
+{
+	vec3 r = orig - center;
+	float height = sqrt(dot(r, r)) - geosphereRadius;
+
+	return height;
+}
+
+void scatter(out vec2 density, const in vec3 orig, const in vec3 center)
+{
+	float height = height(orig, center);
+
+	density = -height / baricStep;
+
+	// earth atmospheric density: 1.225 kg/m^3, divided by 1e5
+	// 1/1.225e-5 = 81632.65306
+	float earthDensities = geosphereAtmosFogDensity * 81632.65306f;
+	density /= earthDensities;
+}
+
+void findClosestHeight(out float h, out float t, const in vec3 orig, const in vec3 dir, const in vec3 center)
+{
+	vec3 radiusVector = center - orig;
+	vec3 tangent = dot(dir, radiusVector) * dir;
+	vec3 normal = radiusVector - tangent;
+	h = sqrt(dot(normal, normal));
+	t = dot(tangent, dir);
+}
+
+// error function
```

Comment:
This should be described in more detail, at minimum something along the lines of "error function to approximate (in/out)-scattering integral over the ray between the sample position and the light source".